### PR TITLE
chore(client): Added a first prototype of check version

### PR DESCRIPTION
**Pull request by Dracks**
rel #26 

Is not having an auto-updated, but at least is adding a check if there is a new version :) 

Current screenshots:
image
image

## How to test

1. Modify the label Version in the main menu with some version name like v0.6.0
2. Run the game

### CODE REVIEW

**Comment by HunterNN** [comment]
I recommend utilizing the global scene `logic/globals/messages.tscn` for incorporating such dialogues, either by using an existing accept dialog or creating another one based on a `Window`. I also suggest reviewing the script attached to that scene.

To replace the button for new version updates: When the game checks for a new update, it should display this dialogue. The check is performed at the start of the game or when the player clicks on the version label. If the user clicks on the version label and the current version matches, the game can also provide a message stating that the latest version is up to date. Should the detected version not be in "v..." format, it should indicate that this isn't a release version hence cannot identify new updates. The dialogue could include an option for accessing the release page in that situation.

Regarding translation keys appearing: Will they be resolved when this PR gets merged?
image

**Comment by HunterNN** [comment]
Also this plugin could be interesting: https://github.com/daenvil/MarkdownLabel

File: client/logic/ui/new_version_checker.gd [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+class Version:
+	var major: int
+	var minor: int
+	var patch: int
+	var valid: bool = false
+
+	static var VERSION_REGEX : RegEx = RegEx.create_from_string("v(?<major>[0-9]+)\\.(?<minor>[0-9]+)\\.(?<patch>[0-9]+)")
+
+	func _init(data: String)->void:
```
Comment (HunterNN) [inline client/logic/ui/new_version_checker.gd:18]: File needs also to be formatted by the formatter. (Missing spaces before and after `->`)

**Review by HunterNN — CHANGES_REQUESTED** [review]

File: client/logic/ui/components/button_to_uri.gd [hunk 2]
```diff
@@ -1,5 +1,7 @@
 extends GameButton
 
+class_name ButtonToUri
```
[the remark this answers is not in the document]
Comment (Dracks) [reply client/logic/ui/components/button_to_uri.gd:3]: I usually use this order, the samples from godot (If I remember correctly) were in this order. I can change it.

File: client/logic/ui/new_version_checker.gd [hunk 3]
```diff
@@ -0,0 +1,111 @@
+extends Node
+
+@export var OWNER: String = "Codename-Recon"
+@export var REPO: String = "Codename-Recon"
+@export var current_version_label: Label
+
+@onready
+var messages: GlobalMessages = $"/root/Messages"
```
[the remark this answers is not in the document]
Comment (Dracks) [reply client/logic/ui/new_version_checker.gd:8]: I didn't knew this way to reference an autoload. Thanks!

File: client/logic/globals/messages.tscn [hunk 4]
```diff
@@ -1,15 +1,221 @@
...
+[/ul]
+[font_size=27]Other changes[/font_size]
+[ul]bla bla bla
+bla bla bla
+[/ul]
+"
+script = ExtResource("2_xrx86")
+markdown_text = "# New version
+* whatever
+* I don't know
```
[the remark this answers is not in the document]
Comment (Dracks) [reply client/logic/globals/messages.tscn:111]: if we don't get the version from the repo, this message should not be shown ever.

[the remark this answers is not in the document]
Comment (HunterNN) [reply client/logic/globals/messages.tscn:111]: Still, it feels a bit unprofessional to use those placeholders. I'd suggest using generic text about the text not loading correctly, like you did with the credits.

[the remark this answers is not in the document]
Comment (Dracks) [reply client/logic/globals/messages.tscn:111]: I will change it, but part of this placeholder was to validate the texts is shown correctly

**Review by HunterNN — APPROVED** [review]

**Pull request merged**